src/dcegm/state_space.py

```diff
@@ -264,3 +270,36 @@ def create_current_state_and_state_choice_objects(
         reshape_current_state_choice_vec_to_mat,
         transform_between_state_and_state_choice_vec,
     )
+
+
+def determine_states_and_state_choices_per_period(
+    state_space, state_choice_space, num_periods
+):
+    """Determine states and state choices per period and save as dictionaries.
```

Comment:
Adjust docstrings: Create dicts of period-specific states and state-choice combinations.